andela/boondocks-bn-backend
# #169257407 Notification for new travel

#### What does this PR do?
- Create pull request for notification of new travel
#### Description of Task to be completed?
- [X] added the get notification controller
- [X] added socket.io for real-time notifications
- [X]  added the cancel email notification endpoint
- [X] added the email notification service
#### How should this be manually tested?
- Navigate to the root folder where you want to copy this repo
- Run `git clone <link to this repo>
- Run `npm i` to install the dependencies
- Run `npx sequelize-cli db:seed:all` to populate the user table with the needed data
-  Run `npm run watch` to   start the server
- Use postman to:
 1. Sign in using a requester using this url `localhost:3000/api/v1/auth/signin`
2. Create a trip
POST
```
{
	"hotelId": 1,
	"type": "return",
	"leavingFrom": "Nairobi",
	"goingTo": "Kigali",
	"travelDate": "2019-12-02",
	"returnDate": "2019-12-02",
	"reason": "Branch visit in kigali",
	"rooms": [15, 18]
}
```
You should  use a valid email so as to get the notification email

3. Get particular line manager email by signing him in and use his token on route  `localhost:3000/api/v1/notification`

#### Any background context you want to provide?
- If you haven't used socket.io before please read about it before reviewing this PR
#### What are the relevant pivotal tracker stories?
[#169257407](https://www.pivotaltracker.com/story/show/169257407)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/52410156/69938434-ffce5500-14e5-11ea-9aa0-6f8ea862b2f9.png)
![image](https://user-images.githubusercontent.com/52410156/70228266-bc315080-175c-11ea-90b4-b84c627ef8a1.png)



#### Questions:
